shaders/src/light_indirect.fs

```diff
@@ -170,6 +204,38 @@ vec3 getReflectedVector(const PixelParams pixel, const vec3 n) {
     return getSpecularDominantDirection(n, r, pixel.roughness);
 }
 
+// This function returns an IBL lookup direction, taking into account the current IBL type (e.g. infinite spherical, finite/local sphere/box).
+vec3 GetAdjustedReflectedDirection(const MaterialInputs material, const PixelParams pixel, const vec3 normal) {
+    if (frameUniforms.iblTechnique == 0u) return getReflectedVector(pixel, normal);
+
+    // intersect the ray ray_pos + t * ray_dir with the finite geometry; done in the coordinate system of the finite geometry
+    vec3 ray_pos = getWorldPosition() + getWorldOffset() - frameUniforms.iblCenter;
+    vec3 ray_dir = normalize(reflect(-shading_view.xyz, normal.xyz));
+
+    vec3  r  = vec3(0.0); // resulting direction
+    float t0 = -1.0f;     // intersection parameter between ray and finite IBL geometry
+    
+    if (frameUniforms.iblTechnique == 1u) {
+        float R2 = frameUniforms.iblHalfExtents.r; // we store the squared radius to shave off a multiplication here
+        float A = 1.0; // in general, this should be dot(ray_dir, ray_dir) but we have just normalized it a couple of lines ago
+        float B = 2.0 * dot(ray_pos, ray_dir);
+        float C = dot(ray_pos, ray_pos) - R2;
+        vec2 roots = solveQuadratic(A, B, C);
+        t0 = getSmallestPositive(roots.x, roots.y);
+    }
+    else if (frameUniforms.iblTechnique == 2u) {
+        vec2 roots = intersectAABB(ray_pos, ray_dir, -frameUniforms.iblHalfExtents, frameUniforms.iblHalfExtents);
+
+        t0 = getSmallestPositive(roots.x, roots.y);
+    }
+
+    // translate results back to world space
+    vec3 intersection_point = ( t0 < 0.0 ) ? getReflectedVector(pixel, shading_normal) : frameUniforms.iblCenter + ray_pos + t0 * ray_dir;
+    r = normalize(intersection_point);
```

Comment:
🤔 So, the adjusted reflected direction is given by the vector pointing from the origin of the finite box/sphere to the intersection point, right? 

Is `intersection_point` in world space, (as the comment leads me to believe)?. If yes, the normalization step seems to be incorrect, we should be doing that in the coordinate system of the finite box/sphere.